yb-voyager/src/srcdb/srcdb.go

```diff
@@ -9,6 +9,7 @@ import (
 
 type SourceDB interface {
 	Connect() error
+	GetConnectionUri() string
```

Comment:
All the callers of the `GetConnectionUri()` are inside the same `srcdb` package. Then why do we need to make it part of a public interface? The "connection management" is an internal implementation detail that should be hidden from the users of `SourceDB`.

Reply:
Actually with these changes there is a now requirement of Get function for URI for pg_dump.
```
[root@ip-10-9-80-202 yb-voyager]# grep -Rn -A 1 -B 1 "source.sourceDB.GetConnectionUri" *
yb-voyager/src/srcdb/pg_dump_export_data.go-40-	cmd := fmt.Sprintf(`pg_dump "%s" --no-blobs --data-only --compress=0 %s -Fd --file %s --jobs %d`,
yb-voyager/src/srcdb/pg_dump_export_data.go:41:		source.sourceDB.GetConnectionUri(), tableListPatterns, dataDirPath, source.NumConnections)
yb-voyager/src/srcdb/pg_dump_export_data.go-42-
--
yb-voyager/src/srcdb/pg_dump_extract_schema.go-19-	cmd := fmt.Sprintf(`pg_dump "%s" --schema-only --no-owner -f %s`,
yb-voyager/src/srcdb/pg_dump_extract_schema.go:20:		source.sourceDB.GetConnectionUri(), filepath.Join(exportDir, "temp", "schema.sql"))
yb-voyager/src/srcdb/pg_dump_extract_schema.go-21-	log.Infof("Running command: %s", cmd)

```

Reply:
You can modify `pgDumpExportDataOffline()` to accept a `connectionURI` as an argument. The function has only one caller in the `PostgreSQL.ExportData()`. Make the caller pass the value of `pg.getConnectionString()` to the `pgDumpExportDataOffline()`.